src/vorta/keyring/secretstorage.py

```diff
@@ -51,7 +51,10 @@ def get_password(self, service, repo_url):
             items = list(collection.search_items(attributes))
             logger.debug('Found %i passwords matching repo URL.', len(items))
             if len(items) > 0:
-                return items[0].get_secret().decode("utf-8")
+                item = items[0]
+                if item.is_locked():  # Some providers lock items until the user manually approves it
+                    item.unlock()
+                return item.get_secret().decode("utf-8")
```

Comment:
This won't work if the user dismisses the unlock. An unlocking mechanism is already implemented for collections here:
https://github.com/borgbase/vorta/blob/8126f4e696ed7bb40b8fc870b1dfc746672d9d11/src/vorta/keyring/secretstorage.py#L57-L66

Reply:
Alright. I will address these issues.

Reply:
I can also take on the remaining issues if you don't have the time.